### DISCUSSION TITLE
Need help / pointer on scenario having a user interactive and machine authentification scenario

### DISCUSSION THREAD

**Discussion by ghost** [OpenIDC/mod_auth_openidc#895] [Q&A] [answered]
Hello,

I'm currently involved in a project that require changing our current integration using Shibboleth to one using Azure B2C.

We have a Angular application talking to a spring boot service behind Apache. There is currently no authentification / ressource protection in the Spring boot service app.  Every path / ressources / API are protected by Shibboleth (using mod_shib in Apache).

To be able to discuss with the architects, validate work to be done I tested a few things including using mod_auth_openidc as a replacement to mod_shib (I configured a basic setup with app / user / login / password reset flow in Azure b2c). Seems to works fine and I definitively see it as an option to consider as we could do the move to Azure with minimal changes.

In our current setup I also support a "machine" access for integration / postman use without interactive login using apache autbasic local accounts. Not very modern but was an acceptable solution with the constraints that we had.

So I'm wondering how if there would be a way to support "client credential grant" using mod_auth_openidc while also having the interactive login user flow (so that I could define machine account in Azure instead of a authbasic apache local account).

I have seen this issue https://github.com/zmartzone/mod_auth_openidc/issues/85 that suggest to me that using another vhost could be a way to do that.

Would there be another way to approach this?

I will also consider moving ressource / api protection on the service side but it's out of the scope of mod_auth_openicd.

I'll read some more and see if I can figure out this by myself but would gladly take any pointers!

Thanks for the great software, documentation is very good and old issues / discussion are also very instructive and helpful.

Eric

**Comment by ghost** [comment]
Found this : https://groups.google.com/g/mod_auth_openidc/c/MjMhb-7VxkI/m/c8aw6RR2CgAJ

**Comment by ghost** [comment, marked as answer]
Also this :  https://github.com/zmartzone/mod_auth_openidc/wiki/Single-Page-Applications#allowing-both-oauth-20-and-openid-connect

**Reply by zandbelt** [reply]
that would be the way to go indeed, where in the non-web/user case the (rich) OAuth client gets its token separately from the Authorization Server using a client credentials grant
